packages/web-components/fast-element/src/components/fast-definitions.ts

```diff
@@ -190,6 +199,29 @@ export class FASTElementDefinition<
         this.styles = ElementStyles.normalize(nameOrConfig.styles);
 
         fastElementRegistry.register(this);
+
+        Observable.getNotifier(this).subscribe(
```

Comment:
Another approach would be to have the controller instances themselves subscribe to the definition. That way, they would be notified when this property changes, and the controller itself could go through template tear-down and rebinding processes. 

I *think* if you took that approach, you wouldn't need the reregister method on the registry and wouldn't need to call `$fastController.connect()` from the definition instance.

Reply:
Brilliant, thanks for the suggestion!